packages/quantic/force-app/main/default/lwc/quanticResultQuickview/quanticResultQuickview.js

```diff
@@ -185,7 +185,9 @@ export default class QuanticResultQuickview extends LightningElement {
     const {pushRecentResult} = this.headless.loadRecentResultsActions(
       this.engine
     );
-    this.engine.dispatch(pushRecentResult(Object.create(this.result)));
+    this.engine.dispatch(
+      pushRecentResult({...this.result, raw: {...this.result.raw}})
+    );
```

Comment:
Ooof I am not sure of this solution...

IIRC we want to create a copy of the result object to make sure the recentResult object cannot modify by reference the result itself.

IIRC we started with first doing a `JSON.parse(JSON.stringify(this.result))` but this was causing lag because of Salesforce's Locker too having to run a function on each of the result properties.

So I think this was changed to `Object.create` which is normally an alternative and a better way to "clone" a JS object without keeping the references.

So no the issue is that `Object.create` when used on the Proxy given to us by Salesforce's Locker is causing issues?

I am afraid that de-structuring the result is not enough to completely kill all the references to the Result Object.

Also do we really need the `recentResult` feature in the Service Console? Is it only for the Hosted Search Page? Because I think we don't use it on purpose for the InsightPanel?

This was the previous bug which led us to use the Object.create
https://github.com/coveo/ui-kit/pull/3016

Really not sure this is the fix for this...

Reply:
Actually, our goal in the past we have done JSON.parse(JSON.stringify(...)) or Object.Create() is not to delete the references to this object but simply because Headless always find difficulties dealing with Proxy Object instead of a normal Object.
So all these procedures had one goal which is converting the Proxy to a normal Object.

The problem with `JSON.parse(JSON.stringify(...))` is that it became super slow at a certain moment, probably because SF changed how they do their transfer from an object to a proxy.
We replaced it with Object.Create()` that worked well for a while but it's no longer working as expected now, doing for example Object.Create() of a Proxy is now returning an empty object which break the recent queries feature.
and Again I think `Object.Create()` stopped working because SF changed the logic they use to build a Proxy out of an Object.

Finally, de-structuring seems now like the method that is doing the job properly, the job of converting a Proxy to an object, which means that Headless is working as expected and the recent queries are working as expected with no errors that were reported for this issue.

Also, this issue is not happening only in the Service Console, it happens everywhere in Salesforce, even in the Community the same error was noticed and the same recent results issue was noticed.

So basically, that's the logic behind it, it's not about killing the references to a Result object, but it is about avoiding passing a Proxy to Headless.

Reply:
Ok well put a comment in the code describing what to do the next time Salesforce changes something that breaks this new spread ;)